### ISSUE TITLE
[#24] experiment: LoRA fine-tuning for face preservation with Flux2CLI

### ISSUE THREAD

**Issue by WoojinAhn** [WoojinAhn/local-llm-pipeline#25]
## Summary

FLUX.2 i2i (\`--images\` visual conditioning) does not preserve specific faces — verified with Son Heung-min test in #24. The only reliable way to reproduce a specific person's face locally is **LoRA fine-tuning** via Flux2CLI's \`train-lora\` subcommand.

## Feasibility Check (M5 Max 128GB)

| Model | VRAM needed | Training time (20 images, 1000 steps) | Quality |
|-------|------------|---------------------------------------|---------|
| Klein 4B | ~8 GB | ~1-2 hours | Good (prototype) |
| **Klein 9B** | ~18 GB | ~3-5 hours | **Recommended** |
| Dev 32B | ~50-70 GB | ~10-20 hours | Best (overkill) |

128GB is more than sufficient. Bottleneck is dataset preparation and training time, not hardware.

## Dataset Requirements

- 15-30 photos of target person (varied angles, expressions, lighting)
- Each photo paired with a caption file (txt or jsonl)
- Recommended 1024x1024 resolution
- Mix of close-up face, half-body, full-body shots
- \`[trigger]\` placeholder in captions gets replaced with a unique word (e.g., "sheungmin")

## Example Caption

\`\`\`
son-01.jpg → "[trigger] Asian male soccer player, short black hair, smiling, white background"
son-02.jpg → "[trigger] running on soccer field, red jersey number 7, action shot"
\`\`\`

## Scope

- [ ] Collect dataset (~20 Son Heung-min photos)
- [ ] Write caption files
- [ ] Configure training YAML (Klein 9B recommended)
- [ ] Run training via \`./bin/Flux2CLI train-lora\`
- [ ] Test generated LoRA with validation prompts
- [ ] Document final configuration in repo

## CLI Reference

\`\`\`bash
./bin/Flux2CLI train-lora datasets/son \\
  --model klein-9b \\
  --rank 32 \\
  --alpha 32 \\
  --learning-rate 1e-4 \\
  --epochs 10 \\
  --gradient-checkpointing \\
  --trigger-word sheungmin \\
  --output outputs/lora-son.safetensors
\`\`\`

## Notes

- This is the only reliable way to preserve specific person's face locally with FLUX.2
- Commercial alternatives (Midjourney, DALL-E 3) reproduce celebrities without training but are not local
- Consider whether the training effort is worth the outcome for one-off images vs building a reusable LoRA

**Comment by WoojinAhn** [comment]
Closing on the effectiveness gate: prerequisite unmet. LoRA face preservation sits behind the FLUX chain (#21 -> #23), and #23 (/generate in multimodal.py) is still unimplemented — verified, no flux/generate code in multimodal.py. Parking it open just adds backlog noise. Reopen when the FLUX path actually moves.
